ios/brave-ios/Sources/PlaylistUI/PlaylistItemView.swift

```diff
@@ -0,0 +1,200 @@
+            .font(.callout.weight(.semibold))
+            .foregroundColor(Color(braveSystemName: .textPrimary))
+            .frame(maxWidth: .infinity, alignment: .leading)
+          if isSelected {
+            LeoPlayingSoundView(isAnimating: isPlaying)
+              // FIXME: Should this scale? Its just cosmetic
+              .frame(width: 16, height: 16)
+              .foregroundStyle(
+                isPlaying
+                  ? Color(braveSystemName: .primary50) : Color(braveSystemName: .iconDisabled)
+              )
+          }
+        }
+        HStack(alignment: .firstTextBaseline) {
+          Text(duration, format: .time(pattern: .minuteSecond))
+          if let downloadState {
+            switch downloadState {
+            case .downloading(let percentCompleted):
+              Label {
+                Text("Preparing")
+              } icon: {
+                Gauge(value: percentCompleted) {
+                  EmptyView()
+                }
+                .gaugeStyle(.circularCapacity(lineWidth: gaugeLineWidth))
+                .frame(width: progressViewSize, height: progressViewSize)
+              }
+              .foregroundColor(Color(braveSystemName: .primary50))
+            case .completed:
+              Label {
+                // FIXME: Better accessibility label
+                Text("Item Ready")
+              } icon: {
+                Image(braveSystemName: "leo.check.circle-outline")
+              }
+              .labelStyle(.iconOnly)
+            }
+          }
+        }
+        .font(.footnote)
+        .foregroundStyle(Color(braveSystemName: .textSecondary))
+      }
+      .frame(maxWidth: .infinity, alignment: .leading)
+    }
+    .padding(.horizontal, 12)
+    .padding(.vertical, 8)
+  }
+}
+
+// FIXME: Replace with a TimelineView variant if possible
+struct LeoPlayingSoundView: View {
+  var isAnimating: Bool
+
+  @State private var barHeights: SIMD4<Double> = .init(x: 0.1, y: 0.1, z: 0.1, w: 0.1)
+
+  private func randomizeBarHeights() {
+    barHeights = .init(
+      x: Double.random(in: 0.2...0.45),
+      y: Double.random(in: 0.3...1),
+      z: Double.random(in: 0.4...0.75),
+      w: Double.random(in: 0.5...0.9)
+    )
+  }
+
+  var body: some View {
+    LeoPlayingSoundShape(barHeights: barHeights)
+      .animation(.linear(duration: 0.3), value: barHeights)
+      .onReceive(
+        Timer.publish(every: 0.3, on: .main, in: .default).autoconnect(),
+        perform: { _ in
+          if !isAnimating { return }
+          randomizeBarHeights()
+        }
+      )
+      .onAppear {
+        if isAnimating {
+          randomizeBarHeights()
+        }
+      }
+      .onChange(of: isAnimating) { newValue in
+        if newValue {
+          randomizeBarHeights()
+        } else {
+          barHeights = .init(
+            x: 0.1,
+            y: 0.1,
+            z: 0.1,
+            w: 0.1
+          )
+        }
+      }
+  }
+
+  struct LeoPlayingSoundShape: Shape {
```

Comment:
A little doc to explain what actually does will be actually helpful for SoundShape

Reply:
I'll add a FIXME to add doc here

Reply:
added doc in https://github.com/brave/brave-core/pull/23425/commits/e6d4680f8eaafb469f1d2c337d9b91fbfeca98fb